hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -227,13 +246,20 @@ public HoodieTableConfig() {
 
   private void fetchConfigs(FileSystem fs, String metaPath) throws IOException {
     Path cfgPath = new Path(metaPath, HOODIE_PROPERTIES_FILE);
+    Path backupCfgPath = new Path(metaPath, HOODIE_PROPERTIES_FILE_BACKUP);
     try (FSDataInputStream is = fs.open(cfgPath)) {
       props.load(is);
+      // validate checksum for latest table version
+      if (HoodieTableVersion.current().equals(getTableVersion()) && !isValidChecksum()) {
```

Comment:
can we make this getTableVersion >= HoodieTableVersion.FOUR so we don't need to fix this going forward

Reply:
Oh yeah, done!